SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -189,6 +189,22 @@ UnitTest {
 		}
 	}
 
+	testForException { |func, errorClass|
+		var test = if(errorClass.isNil) { _.notNil } { _.isKindOf(errorClass.asClass) };
+		var result = false;
+		func.try { |error| result = test.(error) };
+		^result
```

Comment:
There are three possible outcomes here - there was no throw, there was a throw and the error types matched, and there was a throw and the error types didn't match. The test reporting system needs to disambiguate all three cases. With that in mind, perhaps a better way to structure this is by having `testForException` return either the exception or `nil`. For `assertNoException`, not much changes, but in `assertException` both failure conditions (no throw / wrong type) need to be handled, and I suppose you would add that info through `details`?

Reply:
I'm doing two asserts there, that is better because it is more explicit.